### ISSUE TITLE
Add `paddle.nn.functional.loss` NN Loss Functions to PaddlePaddle Frontend

### ISSUE THREAD

**Issue by hmahmood24** [unifyai/ivy#15053]
Add [NN Loss Functions](https://www.paddlepaddle.org.cn/documentation/docs/en/api/index_en.html) to PaddlePaddle Frontend

> Please keep in mind that the proper way to link an issue to this list is to comment "- [ ] #issue_number" while the issue's title only includes the name of the function you've chosen.

- [ ] binary_cross_entropy
- [ ] binary_cross_entropy_with_logits
- [ ] cross_entropy
- [ ] dice_loss
- [ ] hsigmoid_loss
- [ ] kl_div
- [ ] l1_loss
- [ ] log_loss
- [ ] margin_ranking_loss
- [ ] mse_loss
- [ ] nll_loss
- [ ] npair_loss
- [ ] sigmoid_focal_loss
- [ ] smooth_l1_loss
- [ ] softmax_with_cross_entropy
- [ ] margin_cross_entropy
- [ ] square_error_cost
- [ ] ctc_loss
- [ ] hinge_embedding_loss
- [ ] cosine_embedding_loss
- [ ] multi_label_so
- [ ] triplet_margin_with
- [ ] tripl
- [ ] so

**Comment by zayed-haque** [comment]
- [ ] #15097

**Comment by diksha-shrivastava13** [comment]
- [ ] #15950

**Comment by RohitPawar2406** [comment]
- [ ] #16455

**Comment by mobley-trent** [comment]
- [ ] #16727

**Comment by KevinUli** [comment]
- [ ] #16741

**Comment by KevinUli** [comment]
- [ ] #16742

**Comment by Kamil2601** [comment]
- [ ] #17400

**Comment by m-gopichand** [comment]
#17425

**Comment by ligia26** [comment]
#18231

**Comment by peterwisu** [comment]
#18373

**Comment by metehanberker** [comment]
margin_ranking_loss https://github.com/unifyai/ivy/issues/18375

**Comment by peterwisu** [comment]
log_loss #18515

**Comment by metehanberker** [comment]
hinge_embedding_loss: https://github.com/unifyai/ivy/issues/19010

**Comment by ZakariaR1ad** [comment]
- [  ] #19271

**Comment by omojolasamson** [comment]
- [ ] #19335

**Comment by rajkumar-b** [comment]
- [ ] #19471

**Comment by erimdogan** [comment]
softmax_with_cross_entropy #19546

**Comment by BranoKusik** [comment]
Hi @hmahmood24 , can you assign this task for me?

**Comment by Davidch4r** [comment]
- [ ] #19759

**Comment by Davidch4r** [comment]
- [ ] #19884

**Comment by Indraneel99** [comment]
- [ ] #20797

**Comment by ser-art** [comment]
@hmahmood24 hello, could you please add my issue #20552

**Comment by ajay-vishnu** [comment]
- [ ] #20859

**Comment by ajay-vishnu** [comment]
- [ ] #20862

**Comment by bundleabundance** [comment]
I am doing the tripl function

**Comment by masterismail** [comment]
- [ctc_loss] #20976

**Comment by Juligee** [comment]
[ ] #21258

**Comment by aks1204** [comment]
nll_loss #21312"

**Comment by MuhammadTalha-10** [comment]
hsigmoid_loss#20859

**Comment by Praneet9** [comment]
- [ ] #20797

**Comment by masterismail** [comment]
- [ ] #21478

**Comment by Krish2002** [comment]
@hmahmood24 Hey , can i pickup dice loss , sigmoid focal loss ?

**Comment by Seckrel** [comment]
margin_cross_entropy #21759

**Comment by asuzukosi** [comment]
- [ ] #21862

**Comment by rohinish404** [comment]
- [ ] #22052

**Comment by DOnlyRaven** [comment]
#22059

**Comment by HamzaRamzan65** [comment]
[square_error_cost] #22062

**Comment by arshPratap** [comment]
- [ ] #22215

**Comment by rohinish404** [comment]
- [ ] #22052

**Comment by arshPratap** [comment]
- [ ]  #22217

**Comment by arshPratap** [comment]
- [ ] #22218

**Comment by zeus2x7** [comment]
ctc_loss #22608

**Comment by BebehCodes** [comment]
- [ ] #22909

**Comment by nme998** [comment]
hsigmoid_loss #15053

**Comment by nme998** [comment]
hsigmoid_loss #23196

**Comment by retrogtx** [comment]
#23398

**Comment by kuchiPie** [comment]
- [] #23705

**Comment by SamruddhiNavale** [comment]
- [ ] #24171

**Comment by SamruddhiNavale** [comment]
- [ ] #24171

**Comment by abdul-rehman18** [comment]
#25769

**Comment by eliaz5536** [comment]
hsigmoid_loss #23196

**Comment by ElPatatone** [comment]
- [ ] #26076

**Comment by Aryan8912** [comment]
hey @KevinUli can you assign to me so I can work on this issue

**Comment by Aryan8912** [comment]
Please check my PR

**Comment by gsai29** [comment]
- [ ] #26076

**Comment by bangaradi** [comment]
- [ ] #26688

**Comment by Amirbn73** [comment]
npair_loss #26696

**Comment by Amirbn73** [comment]
[ ] #26696
